### ISSUE TITLE
File not found error when rendering Quarto to viewer pane with Sync Editor turned on

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#14277]
### System details

    RStudio Edition : Desktop (didn't try server but probably also happens there)
    RStudio Version : 2024.04.0-hourly+456
    OS Version      : any
    R Version       : any

### Steps to reproduce the problem

NOTE: Noticed while testing https://github.com/rstudio/rstudio/issues/14275.

- File / New Project / New Directory / Quarto Project
- Select "Preview in Viewer Pane" via gear icon
- click Render
- Turn on the "Sync Editor" checkbox
- click Render

### Describe the problem in detail

- Get an error dialog saying "Error while opening file"

<img width="305" alt="screenshot of error-while-opening-file dialog" src="https://github.com/rstudio/rstudio/assets/10569626/1d050124-05eb-469e-81e5-5752f0007365">

### Describe the behavior you expected

- No error dialog
- "Sync Editor" feature works (I don't know how to test that)

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
This is a Quarto bug whose fix didn't make it into the patch release. The request log has:

```
Request ID: 2146866974

== REQUEST ======
{
  "method": "open_document",
  "params": [
    "~/scratch/quarto-example/<%- inputFile %>",
    "text",
    "UTF-8"
  ],
  "clientId": "3bcd46e9-c8ad-42b7-a549-01cce00e3948",
  "clientVersion": "71e2137860455805d2fb80a60c23fc44e0aa1b07"
}

== RESPONSE ======
{
  "error": {
    "code": 100,
    "message": "Error occurred while executing method",
    "error": {
      "code": 2,
      "category": "system",
      "message": "No such file or directory",
      "location": "Error rstudio::session::modules::source::(anonymous namespace)::openDocument(const json::JsonRpcRequest &, json::JsonRpcResponse *) src/cpp/session/modules/SessionSource.cpp:317"
    }
  }
}
```

And note that the file being opened has a template entry which has not been expanded.

> <%- inputFile %>

See also: https://github.com/quarto-dev/quarto-cli/issues/8459

**Comment by ronblum** [comment]
@ronblum to follow up with Quarto team.

**Comment by ronblum** [comment]
@cscheid

**Comment by kevinushey** [comment]
Possibly fixed by https://github.com/rstudio/rstudio/pull/14374?

**Comment by cscheid** [comment]
Thanks, @ronblum. Is this one something that Quarto should be doing better?

**Comment by kevinushey** [comment]
@cscheid I believe this is related to that other issue with the unexpanded EJS template entry in Quarto, but we also have a workaround in RStudio for Quarto 1.4 now so either way I think we are good to go.

**Comment by cscheid** [comment]
Sorry, I still don't quite understand. More directly:

if the EJS template bug is fixed (which it is on 1.5 and backported to v1.4 for our next patch release), then will that have fixed your bug? In other words: does Quarto need to do anything else that we haven't?

**Comment by kevinushey** [comment]
Sorry; to be explicit -- I don't think we need anything else on the Quarto side.
